From d37d1c82be28c1bb3daad74fc53f536071471404 Mon Sep 17 00:00:00 2001
From: AmeChanRain <downloadbot007@gmail.com>
Date: Thu, 19 Jan 2023 19:06:12 +0800
Subject: [PATCH 3/3] SystemUI: biometrics: Add Xiaomi UDFPS Display Mode
 Provider impl

[AngelaCooljx]: Modifiy from Add Nothing UDFPS HBM provider impl, adapt
to Xiaomi

Co-authored-by: Kujou Yuko <i@779.moe>
Change-Id: If5c03b38785a2571d214cf323efbc1bd1cfd0e83
Signed-off-by: Kujou Yuko <i@779.moe>
---
 .../XiaomiUdfpsDisplayModeProvider.kt         | 40 +++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 packages/SystemUI/src/com/android/systemui/biometrics/XiaomiUdfpsDisplayModeProvider.kt

diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/XiaomiUdfpsDisplayModeProvider.kt b/packages/SystemUI/src/com/android/systemui/biometrics/XiaomiUdfpsDisplayModeProvider.kt
new file mode 100644
index 000000000000..5d17c00dcb36
--- /dev/null
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/XiaomiUdfpsDisplayModeProvider.kt
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2022 Project Kaleidoscope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.android.systemui.biometrics
+
+import android.content.Context
+import android.os.FileUtils
+import android.view.Surface
+
+import java.io.IOException
+
+class XiaomiUdfpsDisplayModeProvider constructor(
+    private val context: Context
+): UdfpsDisplayModeProvider {
+
+    override fun enable(onEnabled: Runnable?) {
+        try {
+            FileUtils.stringToFile(DISPPARAM_PATH, "0x20000")
+        } catch (e: IOException) {
+            // Do nothing...
+        }
+        onEnabled?.run()
+    }
+
+    override fun disable(onDisabled: Runnable?) {
+        try {
+            FileUtils.stringToFile(DISPPARAM_PATH, "0xE0000")
+        } catch (e: IOException) {
+            // Do nothing...
+        }
+        onDisabled?.run()
+    }
+
+    companion object {
+        private const val DISPPARAM_PATH = "/sys/class/drm/card0-DSI-1/disp_param"
+    }
+}
-- 
2.39.1

